src/Debugger-Oups/OupsDebugRequest.class.st

```diff
@@ -55,8 +55,15 @@ OupsDebugRequest class >> newForContext: aContext [
 
 { #category : #'instance creation' }
 OupsDebugRequest class >> newForException: anException [
+	"By default we assume that the debugger is requested from the process itself as the result of anException. In that case we must use #realActiveProcess instead of #activeProcess.
+	When debugger opens it must stop the current process. Otherwise the error signal will return to the sender and it will continue execution. Obviously it may cause an unexpected behavior and in worst case it may break the system completely (with some infinite recursion triggering debuggers at every step).
+	Therefore in this scenario we use #realActiveProcess to reliably get a really running process. 
+	For more details see #realActiveProcess and #terminateRealActive comments"
 	<debuggerCompleteToSender>
-	^ self basicNew exception: anException; yourself
+	^ self basicNew 
+			exception: anException;
+			process: Processor realActiveProcess;
+			yourself
```

Comment:
I will leave @StevenCostiou to refactor it later if needed. 
My suggestion is to make the process retrieval an explicit for the users and remove the lazy initialization.